### PR TITLE
LPS-55012

### PR DESCRIPTION
**Pull request by hhuijser**

### CODE REVIEW

**Comment by matethurzo** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by matethurzo** [comment]
Hey @zxdgoal,

I've checked this pull request from code perspective it's fine, but it's introducing a huge regression. With this change the page attributes are becoming editable on the live environment, so the users can go and change different attributes of a page.

The problem is if they change something on the live, with a consequent publish it will be ruined, or even worse they would break a reference on the live that is referencing the page currently being modified. This would lead to inconsistent behavior. Generally the idea is to change on the staging side and then publish to live.

Do you know the reason why they need this information displayed on the live?

Basically even though we display the information, it should not be editable since it's potentially dangerous and they can get themselves into trouble.

Thanks,

Máté

**Comment by zxdgoal** [comment]
Hey  @matethurzo 

I am not sure if you have a look my comment on LPS.

``showButtons="<%= (selLayout.getGroupId() == layoutsAdminDisplayContext.getGroupId()) && SitesUtil.isLayoutUpdateable(selLayout) && LayoutPermissionUtil.contains(permissionChecker, selLayout, ActionKeys.UPDATE) %>"``

Here `showButtons' always return false on live site.

So the save button is not shown.

Consequently, the end user has no way to edit.

As I said, the protection ``test="<%= !selGroup.hasLocalOrRemoteStagingGroup() || selGroup.isStagingGroup()`` is not necesseary.

Thanks
John.

**Comment by zxdgoal** [comment]
Hey @matethurzo 

Not sure if you have time to see my last comment. This is a issue from our customer. 

Sorry for push. 

Thanks
John

**Comment by matethurzo** [comment]
hey,

let me check the pull request again.

**Comment by matethurzo** [comment]
Hey @zxdgoal,

I've checked the pull request again and I've been doing some background research too.

(1) First of all I'd like to know why the customer wants to see these information on live? If they cannot modify it, then what's the point?

(2) I've been talking with Julio and we have agreed that this is the expected behavior on 6.2. The basic workflow is the following: They edit something on stage and they can view everything there. Also on remote live usually users cannot access the control panel, so these information would be hidden for them for sure.

(3) Even though the save button is not visible this change causes an inconsistent behavior. I can modify different values of the pages, but I cannot save it. This is not user friendly, and I think this will end up in more tickets saying that they want to edit something but they cannot save it.
Furthermore there are page JS based validations, where the user can cause inconsistent behavior, and there would be a validation problem due to the changed values.

We need to know why they want to see these information. If they want to change it on live we are not going to allow that in 6.2 for sure.

Thanks,

Máté

**Comment by zxdgoal** [comment]
Hey @matethurzo 

Ok, let me stand on the customer's ground for this issue.

1) We used to SEE these details about this page on live site and we get accustomed to this function , Why they are removed?  I did not see any good reason of removing it.

The logic of if I am not allowed to modify it, I cannot see it seems not reasonable. Why we are not allowed to SEE page details on live site.

2) On remote site, users, at least this customer does go to control panel, you can check this with Julio, I had a previous ticket with him about site admin in live site. He should remember.

3) Going to another side for re-designing is Ok. But try to make everywhere consistent, At lease in live site, click on public page, I can see some details without save button. But if this direction is too far from the current, please take our customer's feel in consideration.

Personally, I believe the developer who added the redundant protection hasn't not see the button is hidden. The code is added by mistake. We can check this.

Thanks
John.

**Comment by zxdgoal** [comment]
Hey @matethurzo 

Could we reach consensus about this issue?

Thanks
John.

**Comment by matethurzo** [comment]
Hey @zxdgoal,

(1) Basically there is no reason to see these values on live. The workflow is that you are making changes on the stage site and you are publishing everything to live. Live is for the average users, live is where you use the portal, and generally you do not want to see these values. The question is the same, why do they want to see these values? What does it provide to them?

**Comment by zxdgoal** [comment]
Hey  @matehurzo

The admin can also login the live site, our customer want to login the live site to see the page details to make sure the setting published are exactly the same they configured in staging site.

I do understand the concerns about modifying the details on live site.

But I cannot understand your concerns about making the customers SEE the details on live site.

Also, if this is your design( not seeing the details). it looks like this design is inconsistent with other component.

For example, for web content, we can still see its content on live site.

I think allowing end users to see details about pages will not cause any problem.

Thanks
John.

**Comment by matethurzo** [comment]
Hey,

Okay I still don't see any reason to do it, but let's show them. But only if the fields are disabled. If any of the fields is editable it will lead to confusion plus there can be problems with the JS validation. Also there has to be a message stating it cannot be edit on live, because the changes has to be done on stage.

Thanks

**Comment by zxdgoal** [comment]
Hey Mate

I agree with your design and indeed it is more user-friendly.

But before those feature implemented, because of the commit I want to revert, the portal now is inconsistent as clicking on the public page link on Pages/site admin, I can see the details with buttons hidden, but clicking on one specific public page, the details is hidden.

That commit I want to revert should reside in your feature request or story ticket if possible, not solely in an LPS, making end user getting inconsistent behaviour and us hard to explain it with the customer.

I meant before your features are implemented, Can we make portal consistent in this issue?

Thanks
John.

**Comment by matethurzo** [comment]
Hey,

I'm not sure if I completely understood you response, but if the staging check in the JSP is reverted, the attributes will be visible and the user should not be able to modify it. All the fields should be disabled, otherwise it would lead to confusion and different validation problems if they are modifying the values either on purpose or by accident.

This should be part of the fix as well.

Thanks,

Máté

**Comment by zxdgoal** [comment]
Hey Mate

I think your improvement makes sense. We'd better create a story where we document your ideas  I have delivered a workaround for this issue to the customer since the UI is inconsistent for the end user. 

Currently, I have no time working on your improvement.  And my second commit is actually an independent fix. I will resent it. 

Could you please this pr?

/cc @hhuijser, let me know if you have different opinions 

Thanks
John
